packages/components/src/components/tag/tag-utils.ts

```diff
@@ -16,6 +17,29 @@ export const hasInvertedThemeColor = (tagColor: TagColor, theme: Theme): boolean
   const isDark = isThemeDark(theme);
   return (
     (!isDark && tagColor === 'neutral-contrast-high') ||
-    (isDark && tagColor !== 'background-surface' && tagColor !== 'background-default')
+    (isDark &&
+      tagColor !== 'background-surface' &&
+      tagColor !== 'background-default' &&
+      tagColor !== 'notification-information' &&
+      tagColor !== 'notification-warning' &&
+      tagColor !== 'notification-success' &&
+      tagColor !== 'notification-error')
   );
 };
+
+export const getThemedBackgroundHoverColor = (tagColor: TagColor, themedColors: ThemedColors, theme: Theme): string => {
+  const isDark = isThemeDark(theme);
+  const colorMap: { [key in TagColor]: string } = {
+    'background-default': isDark ? themedColors.backgroundColorLighten : themedColors.backgroundColorDarken,
```

Comment:
Couldn't the condition evaluated once instead of 10 times?  
Or the maybe even easier would be to interpolate the key, something like
```
const keySuffix = isDark ? 'Darken' : 'Lighten';
...
  'neutral-contrast-high': themedColors[`contrastHighColor${keySuffix}`]

```

Reply:
Changed it as you described, with one exception (contrast high color)